### PR TITLE
Travis build stages

### PR DESCRIPTION
**Pull request by mattbennett**
Use build stages to avoid performing static analysis in every environment

### CODE REVIEW

**Comment by davidszotten** [comment]
are pyflakes and pylint py2/3 independent?

**Comment by mattbennett** [comment]
I assume so, although not really sure. We could easily run both 2x and 3x in parallel.

**Comment by mattbennett** [comment]
Looks like it does make a difference (https://stackoverflow.com/questions/23052637/specify-which-python-version-pylint-should-evaluate-for) so I've added another step to the static build stage

File: .travis.yml [hunk 1]
```diff
@@ -28,46 +28,72 @@ cache:
 
 matrix:
   include:
-    - python: 2.7
+    - stage: static
+      python: 2.7
+      env: TOX_ENV=py27-static
+    -
```
Comment (davidszotten) [inline .travis.yml]: not sure i understand the convention of blank lines in here

Comment (mattbennett) [reply .travis.yml]: It is a bit weird. Was following https://blog.travis-ci.com/2017-05-11-introducing-build-stages .

Re-reading that suggests that they're _optional_ though, so I will put them back in again.

**Pull request merged**